tests/e2e/test_runtime.py

```diff
@@ -224,3 +224,62 @@ def test_terminate_process_sigint():
             process.kill()
 
     assert process.returncode == 130
+
+
+@pytest.mark.e2e
+def test_hot_reload():
+    """
+    Execute a rulebook with hot-reload option,
+    check for first action being triggered,
+    then modify the content of the rulebook,
+    check for the other action being triggered.
+    """
+
+    rulebook = utils.BASE_DATA_PATH / "rulebooks/test_hot_reload.yml"
+    cmd = utils.Command(rulebook=rulebook, hot_reload=True)
+
+    LOGGER.info(f"Running command: {cmd}")
+
+    process = subprocess.Popen(
+        cmd,
+        cwd=utils.BASE_DATA_PATH,
+        text=True,
+        stdout=subprocess.PIPE,
+    )
+
+    fin = open(rulebook, "rt")
+    original_data = fin.read()
+    fin.close()
```

Comment:
It is fine, but there is a better a more pythonic way to do it with a context manager:
```suggestion
    with open(rulebook, "rt") as file:
        original_data = file.read()
```

In that way you ensure the file is properly closed even if there is some error.  

Also you can do it as well in oneline if you use the [pathlib library](https://docs.python.org/3/library/pathlib.html) with the [read_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) method.